docs/deployment/configuration/resource_manager.rst

```diff
@@ -0,0 +1,113 @@
+.. _deployment-configuration-resource-manager:
+
+#####################
+Flyte ResourceManager
+#####################
+
+**Flyte ResourceManager** is a configurable component that allows plugins to manage resource allocations independently. It helps track resource utilization of tasks that run on Flyte. The default deployments are configured as ``noop``, which indicates that the ResourceManager provided by Flyte is disabled and plugins rely on each independent platform to manage resource utilization. In situations like the K8s plugin, where the platform has a robust mechanism to manage resource scheduling, this may work well. However, in a scenario like a simple web API plugin, the rate at which Flyte sends requests may overwhelm a service and benefit from additional resource management.
+
+The below attribute is configurable within FlytePropeller, which can be disabled with:
+
+.. code-block:: yaml
+
+    resourcemanager:
+      type: noop
+
+The ResourceManager provides a task-type-specific pooling system for Flyte tasks. Optionally, plugin writers can request resource allocation in their tasks.
+
+A plugin defines a collection of resource pools using its configuration. Flyte uses tokens as a placeholder to represent a unit of resource.
+
+How does a Flyte plugin request for resources?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Flyte plugin registers the resource and the desired quota of every resource with the **ResourceRegistrar** when setting up FlytePropeller. When a plugin is invoked, FlytePropeller provides a proxy for the plugin. This proxy facilitates the plugin's view of the resource pool by controlling operations to allocate and deallocate resources.
+
+.. dropdown:: :fa:`info-circle` Enabling Redis instance
+   :title: text-muted
+   :animate: fade-in-slide-down
+
+   The ResourceManager can use a Redis instance as an external store to track and manage resource pool allocation. By default, it is disabled, and can be enabled with:
+
+   .. code-block:: yaml
+
+       resourcemanager:
+          type: redis
+          resourceMaxQuota: 100
+          redis:
+            hostPaths:
+              - foo
+            hostKey: bar
+            maxRetries: 0
+
+Once the setup is complete, FlytePropeller builds a ResourceManager based on the previously requested resource registration. Based on the plugin implementation's logic, resources are allocated and deallocated.
+
+During runtime, the ResourceManager:
+
+#. Allocates tokens to the plugin.
+#. Releases tokens once the task is completed.
```

Comment:
I would move the line "In this manner, Flyte plugins intelligently throttle resource usage during parallel execution of nodes" up here.